cubedash/summary/_stores.py

```diff
@@ -542,14 +543,15 @@ def get_or_update(
         year: Optional[int] = None,
         month: Optional[int] = None,
         day: Optional[int] = None,
+        force_refresh: Optional[bool] = False
     ):
         """
         Get a cached summary if exists, otherwise generate one
 
         Note that generating one can be *extremely* slow.
         """
         summary = self.get(product_name, year, month, day)
-        if summary:
+        if summary and not force_refresh:
```

Comment:
I don't think this will be enough.

The summaries are generated top-down: it summarises a product by grabbing all year summaries (generating if missing), and it summaries a year by fetching all 12 of the month summaries (generating if missing), etc.

So when you force-refresh a year, it will just grab the old, unrefreshed, month summaries to recreate it. (I assume?).

Reply:
ok ;-)